server/controllers/message.controller.js

```diff
@@ -177,7 +178,32 @@ function list(req, res) {
     }
 }
 
-function count() {}
+/**
+ * Return count and message IDs of all messages belonging to owner.
+ * Return count and message IDs of unread messages belonging to owner.
+ * "Owner" will not exist as part of url once this integrates with auth service.
+ */
+function count(req, res) {
+    const queryObject = {};
+    const whereObject = {};
+    if (req.query.option === 'unread') {
+        whereObject.owner = req.user.username;
+        whereObject.readAt = null;
+        whereObject.originalMessageId = { [Op.ne]: Sequelize.col('id') };
+        queryObject.where = whereObject;
+        queryObject.attributes = ['id', 'owner', 'readAt', 'to', 'from', 'originalMessageId'];
+    } else if (req.query.option === 'all') {
+        whereObject.owner = req.user.username;
```

Comment:
Same here. Spread is best practice for object assignment.

Reply:
@jsachs, it doesn't seem like the spread operator used this way will allow you to reassign the previous mappings in `queryObject.where` to a new a new one that includes `whereObject`. If you're looking to concatenate all the `key-value` pairs from `queryObject.where` and `whereObject` and reassign this to `queryObject.where` I believe you'll need `queryObject.where = { ...queryObject.where, ...whereObject };`. However I'm not sure if this is what you're going for

Reply:
With a second look it seems like `queryObject` was an empty object before the assignment so it should be safe.